src/odemis/driver/technolution.py

```diff
@@ -0,0 +1,544 @@
+                tsleep = tnext - tnow
+                if tsleep > 0.01:
+                    time.sleep(tsleep)
+
+                tnext = time.time() + 0.2  # max 5 Hz
+
+        except Exception as e:
+            if command == "terminate":
+                raise Exception("Terminated acquisition")
+
+            elif command is not None:
+                logging.error("last message was not executed, should have perfomed action: '%s'\n"
+                              "received exepction: \n %s" % (command, e))
+
+        logging.debug("Acquisition thread ended")
+
+    def start_acquisition(self):
+        mega_field_data = self._assemble_mega_field_data()
+        self.acq_queue.put(("start", mega_field_data))
+
+    def get_next_field(self, field_num):
+        '''
+        :param field_num: x,y
+        :return:
+        '''
+        self._field_data = FieldMetaData(
+                field_num[0] * self.parent._EBeamScanner._shape.value[0],
+                field_num[1] * self.parent._EBeamScanner._shape.value[1])
+        self.acq_queue.put(("next", self._field_data, self.dataContent.value))
+
+    def stop_acquisition(self):
+        self.acq_queue.put(("stop", None))
+
+    def acquire_single_field(self, field_num=(0, 0)):
+        mega_field_data = self._assemble_mega_field_data()
+        self.acq_queue.put(("start", mega_field_data))
+        self._field_data = FieldMetaData(field_num[0] * self.parent._EBeamScanner._shape.value[0],
+                                         field_num[1] * self.parent._EBeamScanner._shape.value[1])
+        self.acq_queue.put(("next", self._field_data, self.dataContent.value))
+        self.acq_queue.put(("stop", None))
+
+    def terminate(self):
+        def clear_queue(queue_size):
+            if queue_size > 1:
+                self.acq_queue.get()
+                clear_queue(self.acq_queue.qsize())
+            else:
+                logging.info("Cleared the remainder of the acquisition queue upon termination")
+                return
+
+        clear_queue(self.acq_queue.qsize())
+        self.acq_queue.put(("terminate", None))
+
+    def convert_field_num2pixels(self, field_num, mega_field_data):
+        mega_field_data = self._megafield_metadata
+        return (field_num[0] * self._shape.value[0] * mega_field_data.x_eff_cell_size,
+                field_num[1] * self._shape.value[1] * mega_field_data.y_eff_cell_size)
+
+    def _getMetadata(self):
+        """
+        Create dict containing all metadata from siblings and own metadata
+        """
+        md = {}
+        for md_dev in self.md_devices:
+            for key in md_dev.keys():
+                if key not in md:
+                    md[key] = md_dev[key]
+                elif key in (model.MD_HW_NAME, model.MD_HW_VERSION, model.MD_SW_VERSION):
+                    md[key] = ", ".join([md[key], md_dev[key]])
+        return md
+
+    def _data_content2value(self, content):
```

Comment:
@Sabrina85  what do you think of that? I think this was the option I implemented before which you didn't like for possible extensions of these options.

Reply:
Then I guess I am overruled ;) I don't like it so much because we still have the third option "empty". I find it more elegant to have one function which deals with all 3 options. If we use a dict, why not also include `"empty": None`? What do you think @Kleijwegt @pieleric

Reply:
The dict/function could contain `"empty"` too. Still you'll have to separate this option from the rest, as in this case, it shouldn't call at all the ASM to get the content. (Or you put the whole block in a separate function, which immediately returns an empty array if the content is "empty").

Reply:
Indeed. My idea was to have a separate method called in the acquisition thread that checks the contentVA and then returns True, False, or None. And if None, return empty array, else call ASM for image (full or thumbnail).